### DISCUSSION TITLE
Two feature requests for AutoConnectUpdate

### DISCUSSION THREAD

**Discussion by HankLloydRight** [Hieromon/AutoConnect#522] [Feature Request]
Hello Hieromon,
  I have two simple requests for `AutoConnectUpdate` -- neither of which are critical, but would be really nice if you could add.

 First, I noticed that the `otaExtraCaption` option only works with `AutoconnectOTA` and not `AutoConnectUpdate`. It would be nice if we could also add the extra caption to the `AutoConnectUpdate` page.

 Second, along the same lines, when using `AutoConnectUpdate`  with the python update server, occasionally the ESP32 won't be able to retrieve the JSON from the update server and either times-out or says "no firmwares available".  With `AutoconnectOTA`, there's a **"Select File -- [choose file]"** option to load the firmware from the local browser.  It would be nice if when using `AutoConnectUpdate`  we could also get that option to load a firmware file from the local browser as a fall-back when the Update server isn't responding correctly.  Or at least an option or #define directive to enable the local file upload option. 

 Thank you,
-Hank

**Comment by Hieromon** [comment]
@HankLloydRight Sorry to late, I've been spending a time studying and implementing [new aid for AutoConnect](https://github.com/Hieromon/EdgeUnified-eval), which has delayed my catching up on this topic.

Below are the proposed specifications I have considered and would appreciate your input.

1. AutoConnectUpdate should have the similar interface as otaExtraCaption.
    I agree. AutoConnectUpdate will equip **a public member** to hold extra captions like follows:
    ```cpp
    String AutoConnectUpdate::extraCaption
    ```

2. Activation of AutoConnectOTA as an alternative when communication with the update server is unavailable.
    I think all of that could be accomplished by extending the AutoConnectUpdate implementation, but at the cost of forcing a static binding AutoConnectOTA. Rather, I think it would be better to create a new onCatalog exit for AutoConnectUpdate and allow AutoConnectOTA to be instantiated arbitrarily by a sketch at the process of the onCatalog exit. How about you?
